src/cpp/database/database.hpp

```diff
@@ -797,7 +846,7 @@ class Database
     std::atomic<int64_t> next_id_{0};
 
     //! Read-write synchronization mutex
-    mutable std::shared_timed_mutex mutex_;
+    mutable std::recursive_timed_mutex mutex_;
```

Comment:
why this change? The reason of having a shared mutex is to allow several reading operations on parallel, which the recursive mute does not allow for. Is there any chance that the mutex will be locked several times in the same thread now? I do not see in this pull request any new lock operation that may result in a multiple locking.